Fix: Switch to Ubuntu 22.04 for CI workflows

alex-ng-wesoft opened the pull request:
ci.yml currently requests Ubuntu 18.04 runners, but per https://github.com/actions/runner-images/issues/6002 Ubuntu 18.04 runners are now fully unsupported:
```
We have started the deprecation process for Ubuntu 18.04.
[...] the image will be fully unsupported by 2023/04/03
```

Fixes #7.

alex-ng-wesoft committed:
Fix: Switch to Ubuntu 22.04 for CI workflows

ci.yml currently requests Ubuntu 18.04 runners, but per https://github.com/actions/runner-images/issues/6002 Ubuntu 18.04 runners are now fully unsupported:
```
We have started the deprecation process for Ubuntu 18.04.
[...] the image will be fully unsupported by 2023/04/03
```

ADmad commented:
Thanks, could you please update the matrix to run the tests on PHP 8.1 and 8.2 also.

alex-ng-wesoft committed:
Feature: Enable CI for PHP 8.1 and 8.2

alex-ng-wesoft committed:
Feature: Use PHPUnit 8.5 for PHP 8.x CI

PHPUnit 7.5 isn't compatible with PHP 8.1 and PHP 8.2.
This commit adopts the existing strategy used for PHP 8.0 (using
PHPUnit 8.5 instead) and generalises it to all PHP 8.x versions.

alex-ng-wesoft commented:
> Thanks, could you please update the matrix to run the tests on PHP 8.1 and 8.2 also.

Done.
